src/kudu/consensus/leader_election.cc

```diff
@@ -156,53 +156,16 @@ void FlexibleVoteCounter::FetchTopologyInfo(
 
   CHECK(config_.has_commit_rule());
 
-  // A flag to denote if we could resolve a master.
-  // May involve cases of first election in the replicaset,
-  // last known master no longer part of the replicaset etc.
-  bool master_rule_unresolvable = false;
-
-  // Stores the requirement in the master region for a pessimistic
-  // computation of leader election quorum.
-  int master_region_req = -1;
+  // TODO(ritwikyadav) : This function will be simplified (or removed) shortly
+  // with the new quorum specification modes.
+  CHECK(config_.commit_rule().mode() == QuorumMode::SINGLE_REGION_DYNAMIC);
 
   // Step 2: Compute the leader election quorum from the corresponding
-  // data commit quorum. The requirement is for it to intersect with all the
-  // data commit quorums and all the leader election quorums.
-  for (const RegionRuleSpecPB& rule_spec :
-      config_.commit_rule().rule_conjunctions()) {
-    std::string region = rule_spec.region_identifier();
-
-    // This will throw if the configuration is not properly defined.
-    int commit_req_num = ParseCommitRequirement(
-        rule_spec.commit_requirement());
-
-    if (region == "master" && leader_region == "") {
-      master_rule_unresolvable = true;
-      master_region_req = commit_req_num;
-      continue;
-    } else {
-      region = leader_region;
-    }
-
-    // At this point region is resolved and we should be able to
-    // fetch its cardinalities.
-    // TODO(ritwikyadav): It is possible that all voters in a region
-    // have left the replicaset. Handle that case gracefully in a subsequent diff.
-    int total_voters = FindOrDie(*voter_distribution, region);
+  // data commit quorum.
```

Comment:
What about implementing the idea that even for SINGLE_REGION_DYNAMIC case, we would expand the LE quorum to have majority of majority of regions.